firecrown/cosmosis/run.py

```diff
@@ -0,0 +1,207 @@
+import pathlib
+import os
+import sys
+
+# Locate the path to this directory
+# For fiddly reasons that make a great deal of sense in an entirely
+# different context to this one, we pass cosmosis a full path
+# to the interface file
+THIS_DIRECTORY = pathlib.Path(__file__).parent.resolve()
+COSMOSIS_INTERFACE = str(THIS_DIRECTORY.joinpath('interface.py'))
+
+
+def run_cosmosis(config, data):
+    """Run CosmoSIS on the problem.
+
+    This requires the following parameters 'sampler' section
+    of the config:
+
+      sampler - name of sampler to use, e.g. emcee, multinest, grid, ...
+      output - name of file to save to
+
+
+    Parameters
+    ----------
+    config : dict
+        Configuration info, usually read directly from the YAML file
+    data : dict
+        The result of calling `firecrown.config.parse` on an input YAML
+        config.
+
+    """
+    from cosmosis.main import run_cosmosis
```

Comment:
Again import should be at the top unless there is a good reason.

Reply:
There is a good reason - it is to support cases where cosmosis is not installed.
Unless we want to require it?

Reply:
ahhh there is a good reason then. I would prefer a guarded import at the top with try except to make it clear that this is what we are trying to account for.

Reply:
Fixed.